transformer_engine/pytorch/attention.py

```diff
@@ -3246,14 +3247,14 @@ def __init__(
             parameters_split = None
             if not fuse_qkv_params:
                 parameters_split = collections.OrderedDict([
-                    ("query", hidden_size),
-                    ("key", self.hidden_size_kv),
-                    ("value", self.hidden_size_kv),
+                    ("query", attention_hidden_size),
+                    ("key", attention_hidden_size),
+                    ("value", attention_hidden_size),
                 ])
             if self.input_layernorm:
                 self.layernorm_qkv = LayerNormLinear(
                     hidden_size,
-                    hidden_size + 2 * self.hidden_size_kv,
+                    3 * attention_hidden_size,
```

Comment:
```suggestion
                    self.hidden_size_per_attention_head * num_attention_heads + 2 * self.hidden_size_kv
```
@pggPL, we'd still need the separate hidden size (total) for `k` and `v` heads to handle group query attention

Reply:
Here's what we could do alternately:

```
# We still keep this line
self.hidden_size_per_attention_head = kv_channels

# Use `kv_channels` to calculate the `attention_hidden_size`
self.attention_hidden_size = self.hidden_size_per_attention_head * num_attention_heads

# Use `self.attention_hidden_size` to calculate `hidden_size_kv` (which is used in case of GQA)
self.hidden_size_kv = int(self.attention_hidden_size* self.num_gqa_groups // num_attention_heads)

...

LayerNormLinear(
                    hidden_size,
                    self.attention_hidden_size+ 2 * self.hidden_size_kv,
                    ...
)
```

We could probably also rename `attention_hidden_size` to `hidden_size_q` because essentially that's what it's used for in the layer/weight creation below

Reply:
I see the point, but I'm not sure whether using `kv_channels` is best option. Currently `kv_channels` argument does exactly nothing if it is set to correct value. If it is set to incorrect value, the error is raised. Changing the behavior of this argument is an option, but it will be unrelated to the name - we change also `q_channels`.

I propose to add optional argument `attn_hidden_size` - legacy code will work correctly without it. We can leave `kv_channels` in the code - but for example only print some "this argument is deprecated" warning. Or even simply do nothing. I believe (maybe wrongly) that this will not lead to less problems with legacy code, but the argument's names will be more accurate. Let me know what do you think.